packages/neo-one-client-core/src/user/RemoteReceiveUserAccountProvider.ts

```diff
@@ -0,0 +1,70 @@
+import { Block, RawAction, UserAccountProvider } from '@neo-one/client-common';
+import { Endpoint, Message } from './messageTypes';
+
+export interface RemoteReceiveUserAccountProviderOptions {
+  readonly endpoint: Endpoint;
+  readonly userAccountProvider: UserAccountProvider;
+}
+
+export class RemoteReceiveUserAccountProvider {
+  private readonly messageEndpoint: Endpoint;
+  private readonly userAccountProvider: UserAccountProvider;
+  // tslint:disable-next-line readonly-keyword
+  private readonly asyncIterableInstanceMap: { [id: string]: AsyncIterator<Block | RawAction> };
+
+  public constructor({ endpoint, userAccountProvider }: RemoteReceiveUserAccountProviderOptions) {
```

Comment:
In fact, this shouldn't be a class at all, it should just be a function called something like `connectRemoteUserAccountProvider`.